### PR TITLE
fix parse error

### LINKED ISSUE #458: tools/stats setup.bash parse error.
after Setup and Usage in the tools/stats. run
./setup.bash 
get this error:
parse error: Invalid numeric literal at line 1, column 10

System info: Ubuntu 18.04.3 LTS

### PR DESCRIPTION
**Pull request by deaswang**
fix #458

### CODE REVIEW

File: tools/stats/setup.bash [hunk 1]
```diff
@@ -13,7 +13,8 @@ curl -s -XPOST http://admin:admin@$GRAFANA_HOST/api/datasources -H 'Content-Type
 }
 EOF
 
-curl -s -XPOST http://admin:admin@$GRAFANA_HOST/api/dashboards/import -H 'Content-Type: text/json' --data-binary @- << EOF | jq -r "\"http://$GRAFANA_HOST\" + .importedUrl"
+curl -s -XPOST http://admin:admin@$GRAFANA_HOST/api/dashboards/import -H 'Content-Type: text/json' --data-binary @- << EOF
+| jq -r "\"http://$GRAFANA_HOST\" + .importedUrl"
```
Comment (travisperson) [inline tools/stats/setup.bash]: This doesn't run for me as it tries to send this line as part of the body.

What parse error are you getting? Providing your bash version may also help figure out what might be the issue.

I'm using `GNU bash, version 5.0.11(1)-release (x86_64-pc-linux-gnu)`.

We can move the command to the end of the heredoc if that fixes your own parse error as well.

```diff
diff --git a/tools/stats/setup.bash b/tools/stats/setup.bash
index fc920ed..5d9a58e 100755
--- a/tools/stats/setup.bash
+++ b/tools/stats/setup.bash
@@ -13,8 +13,7 @@ curl -s -XPOST http://admin:admin@$GRAFANA_HOST/api/datasources -H 'Content-Type
 }
 EOF
 
-curl -s -XPOST http://admin:admin@$GRAFANA_HOST/api/dashboards/import -H 'Content-Type: text/json' --data-binary @- << EOF
-| jq -r "\"http://$GRAFANA_HOST\" + .importedUrl"
+curl -s -XPOST http://admin:admin@$GRAFANA_HOST/api/dashboards/import -H 'Content-Type: text/json' --data-binary @- << EOF |
 {
   "dashboard": $(cat ./chain.dashboard.json),
   "overwrite": true,
@@ -28,3 +27,4 @@ curl -s -XPOST http://admin:admin@$GRAFANA_HOST/api/dashboards/import -H 'Conten
   ]
 }
 EOF
+jq -r "\"http://$GRAFANA_HOST\" + .importedUrl"
```

Comment (deaswang) [reply tools/stats/setup.bash]: I`m using `GNU bash，version 4.4.20(1)-release (x86_64-pc-linux-gnu)`.
it is default bash version for ubuntu18.04. and only move `|` to new line will parse success. your fix also error.
if only support bash 5. should add dependence require for bash.

Comment (travisperson) [reply tools/stats/setup.bash]: I tried bash `GNU bash, version 4.4.20(1)-release (x86_64-pc-linux-gnu)` (Ubuntu 18.04.3 LTS) and everything still works[0]. Your PR fails to with an error from grafana[1]


[0] https://gist.github.com/travisperson/072635e164333c4f29b7bcb6cd71486a
[1] https://gist.github.com/travisperson/41c9bb08e7e7a634f2067a773765ef41

Can you provide the exact output you get when running the script?

Comment (deaswang) [reply tools/stats/setup.bash]: sorry. it success now. maybe some grafana and influxdb config issue.
close.

**Pull request closed without merging**